src/ecalc/libraries/libecalc/common/libecalc/input/yaml/yaml_models/pyyaml_yaml_model.py

```diff
@@ -125,6 +126,12 @@ def __init__(
                 )
 
         def load(self, yaml_file: ResourceStream):
+            if ComponentNameStr.regex.search(Path(yaml_file.name).stem) is None:
```

Comment:
Some regex magic I don't understand here? What regex pattern is used?

Reply:
https://github.com/equinor/ecalc/blob/830c75e27a29549157658c606e618da381c24e81/src/ecalc/libraries/libecalc/common/libecalc/dto/utils/validators.py#L9

Reply:
Imo you should write which characters that are allowed using the regexp/string we have already

Reply:
ComponentNameStr is the regex we have already, it is imported from validators. Or do I misunderstand your point?

Reply:
The error message here will easily go out of sync with the regex. What if we include $ as acceptable in component name? We will forget to update the error message. Imo there is not really any good way to present this to users (there are libraries that create human friendly descriptions of reg exp tho), but is this an option or sth along those lines:

```
COMPONENT_NAME_ALLOWED_CHARS = 'A-ZÆØÅa-zæøå\d_/\- '
ComponentNameStr = constr(regex=r"^[" + COMPONENT_NAME_ALLOWED_CHARS + "]*$")
...
error_message = f"The eCalc Model filename contains illegal characters. Allowed characters are: {COMPONENT_NAME_ALLOWED_CHARS}"
```

Reply:
Or should we just have a separate regex for the filename, independent of what is allowed for components? I don´t see why they should be identical?

Reply:
It is because we do not have another identificator for the whole eCalc Model (file) other than the filename. In order to reference the total results, we use the filename for that. This is represented as a componet in the results. Alternatives are e.g.:

* could just convert the filename to something that only has accepted characters
* introduce an attribute in the yaml file where use must specify "name of eCalc model"

Reply:
> 'A-ZÆØÅa-zæøå\d_/\- '

I think many users will get confused by the 'A-ZÆØÅa-zæøå\d_/\- '. Isn´t it better to write that the `filename contains illegal special characters` - as long as the filename is included in the error message, the user knows where to look?

Reply:
I agree. No good solution. A bit reluctant to introduce a new big library for "such a small thing as well".

Reply:
https://www.autoregex.xyz/

Reply:
I added the allowed characters, as suggested above. Maybe it is ok, as long as we also communicate that the model file name is wrong - and include the filename in the error message?